### PR TITLE
Add tests on createBindGroupLayout() with new storage texture accesses

### PR DESCRIPTION
**Pull request by Jiawei-Shao**
This patch adds tests on the use of new storage texture accesses `read-only` and `read-write` in the creation of bind group
layouts.

This patch also upgrades the version of @webgpu/types to the latest one to support these two storage texture accesses.

Note that right now this test can only pass when Chromium is launched with `--enable-dawn-features=allow_unsafe_apis`.

Issue: #3078

<hr>

**Requirements for PR author:**

- [*] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [*] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [*] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/api/validation/createBindGroupLayout.spec.ts [hunk 1]
```diff
@@ -163,8 +163,10 @@ g.test('visibility,VERTEX_shader_stage_storage_texture_access')
   .fn(t => {
     const { shaderStage, access } = t.params;
 
+    const appliedSuccess = access ?? 'write-only';
```
Comment (Kangz) [inline src/webgpu/api/validation/createBindGroupLayout.spec.ts:166]: nit: appliedAccess?

File: src/webgpu/api/validation/createBindGroupLayout.spec.ts [hunk 2]
```diff
@@ -436,29 +438,35 @@ g.test('storage_texture,layout_dimension')
...
     `
-  Test that a validation error is generated if the format doesn't support the storage usage.
+  Test that a validation error is generated if the format doesn't support the storage usage. A
+  validation error is also generated if the format doesn't support the 'read-write' storage access
+  when the storage access is 'read-write'.
   `
   )
-  .params(u => u.combine('format', kAllTextureFormats))
+  .params(u =>
+    u.combine('format', kAllTextureFormats).combine('access', kStorageTextureAccessValues)
```
Comment (Kangz) [inline src/webgpu/api/validation/createBindGroupLayout.spec.ts:447]:
```suggestion
    u \\
    .combine('format', kAllTextureFormats) \\
       .combine('access', kStorageTextureAccessValues)
```

**Review by Kangz — COMMENTED** [review]
@ben-clayton I'm surprised that Jiawei had to regenerate the cache here, is that expected?

**Comment by ben-clayton** [comment]
> @ben-clayton I'm surprised that Jiawei had to regenerate the cache here, is that expected?

So the hashes will change if there's files modified that _could_ impact the generation of the cache files. The fact the files have actually changed is likely due to differences in NodeJS versions. [This is discussed here](https://github.com/gpuweb/cts/blob/main/docs/case_cache.md#file-cache):

> Since source changes will sometimes change the hash without changing the generated cache, sometimes the cache will be regenerated unnecessarily. **This is OK, but try to avoid committing no-op regenerations - this will happen if your version of Node produces different gzip outputs than the original committer's Node did for the same input.**

It would be good to try keeping the CTS developers on the same version of the NodeJS, and avoiding no-op regeneration like this.

**Comment by ben-clayton** [comment]
Created https://github.com/gpuweb/cts/pull/3187 to try to prevent continual cache regeneration.

File: src/webgpu/api/validation/createBindGroupLayout.spec.ts [hunk 3]
```diff
@@ -436,29 +438,37 @@ g.test('storage_texture,layout_dimension')
...
-            },
-          ],
-        });
-      },
-      !info.color?.storage
-    );
+    let success = info.color?.storage;
+    if (access === 'read-write' && !info.color?.readWriteStorage) {
+      success = false;
+    }
```
Comment (jzm-intel) [inline src/webgpu/api/validation/createBindGroupLayout.spec.ts:461]: How about
``` suggestion
    const success =
      info.color?.storage && !(access === 'read-write' && !info.color?.readWriteStorage);
```

**Review by jzm-intel — APPROVED** [review]
Looks good to me, thanks

**Review by Kangz — APPROVED** [review]
LGTM, I just don't understand while all these new files where created.

**Pull request merged**